docs/developer/architecture/logs.rst

```diff
@@ -197,6 +197,527 @@ overview page.
+
+.. code-block:: yaml
+
+  serviceMonitor:
+    enabled: true
+    additionalLabels:
+      release: "prometheus-operator"
+
+For Fluent Bit, we will need to define a ``ServiceMonitor`` object:
+
+.. code-block:: yaml
+
+  apiVersion: monitoring.coreos.com/v1
+  kind: ServiceMonitor
+  metadata:
+    labels:
+      app: fluent-bit
+      release: prometheus-operator
+    name: fluent-bit
+    namespace: metalk8s-logging
+  spec:
+    endpoints:
+    - path: /api/v1/metrics/prometheus
+      port: http-metrics
+    namespaceSelector:
+      matchNames:
+      - metalk8s-logging
+    selector:
+      matchLabels:
+        app: fluent-bit
+
+We also need to define alert rules based on metrics exposed by these services.
+This is done deploying new ``PrometheusRules`` object in ``metalk8s-logging``
+Namespace:
+
+.. code-block:: yaml
+
+  apiVersion: monitoring.coreos.com/v1
+  kind: PrometheusRule
+  metadata:
+    labels:
+      release: prometheus-operator
+    name: loki.rules
+    namespace: metalk8s-logging
+  spec:
+    groups:
+    - name: loki.rules
+      rules: <RULES DEFINITION>
+
+There is some recording and alert rules defined in the `Loki repository`_ that
+could be used as a base, then we could enrich these rules later when we will
+have better operational knowledge.
+
+.. _Loki repository: https://github.com/grafana/loki/tree/master/production/loki-mixin
+
+Grafana
+"""""""
+
+To be able to query logs from Loki, we need to add a Grafana datasource,
+this is done adding a ConfigMap ``loki-grafana-datasource`` in
+``metalk8s-monitoring`` Namespace as follows:
+
+.. code-block:: yaml
+
+  apiVersion: v1
+  kind: ConfigMap
+  metadata:
+    name: loki-grafana-datasource
+    namespace: metalk8s-monitoring
+    labels:
+      grafana_datasource: "1"
+  data:
+    loki-datasource.yaml: |-
+      apiVersion: 1
+      datasources:
+      - name: Loki
+        type: loki
+        access: proxy
+        url: http://loki.loki.svc.cluster.local:3100
+        version: 1
+
+To display the logs we also need a dashboard, adding a ConfigMap
+``loki-logs-dashboard`` in Namespace ``metalk8s-monitoring``:
+
+.. code-block:: yaml
+
+  apiVersion: v1
+  kind: ConfigMap
+  metadata:
+    name: loki-logs-dashboard
+    namespace: metalk8s-monitoring
+    labels:
+      grafana_dashboard: "1"
+  data:
+    loki-logs.json: <DASHBOARD DEFINITION>
+
+For the dashboard we will use a view with a simple log panel and variables
+representing labels to filter on.
+Since journal and kubernetes logs will not have the same labels, we could
+either have 2 distinct dashboards or 2 log panels in the same.
+An example of what we want is `Loki dashboard`_.
+
+.. _Loki dashboard: https://github.com/grafana/loki/blob/master/production/loki-mixin/dashboard-loki-logs.json
+
+.. note::
+
+  The ``grafana_datasource: "1"`` and ``grafana_dashboard: "1"`` labels are
+  what is used by the Prometheus Operator to retrieve datasource and dashboard
+  for Grafana.
```

Comment:
I'm not sure they need to be in `metalk8s-monitoring` namespace, I think the Operator has cluster wide rights allowing it to watch/read ConfigMaps.
But I'm not super familiar with ClusterRole and stuff, so maybe I'm misunderstanding something.

Reply:
In this case, it would make sense to store logging-related dashboards/datasources in `metalk8s-logging`

Reply:
Tested, it does not seem to work if outside of the `metalk8s-monitoring` namespace, at least for datasource (it may not be the case for dashboards, but I didn't find a clear answer, I should test it).